smac/intensification/intensification.py

```diff
@@ -112,13 +110,34 @@ def __init__(self, tae_runner: ExecuteTARun,
         self.use_ta_time_bound = use_ta_time_bound
         self.min_chall = min_chall
 
-    def intensify(self, challengers: typing.Optional[typing.List[Configuration]],
-                  optimizer: typing.Optional['SMBO'],
-                  incumbent: Configuration,
-                  run_history: RunHistory,
-                  aggregate_func: typing.Callable,
-                  time_bound: float = float(MAXINT),
-                  log_traj: bool = True) -> typing.Tuple[Configuration, float]:
+        # stage variables
+        # the intensification procedure is divided into 3 'stages':
+        # 1. add incumbent run
+        # 2. race challenger
+        # 3. race against configuration for a new incumbent
+        self.run_incumbent = True
```

Comment:
As there can be three states (run incumbent, run challenger and run default) I think it would be better to use an enum for the different states to make the code more readable.

Reply:
Potentially, there are four stages, with the fourth one being done. I know this can be represented with the current variables, too, but the code is currently very hard to read because the state is encoded across multiple variables.